crates/sui-core/src/authority/shared_object_congestion_tracker.rs

```diff
@@ -149,20 +160,21 @@ impl SharedObjectCongestionTracker {
             return None;
         }
         let start_cost = self.compute_tx_start_at_cost(&shared_input_objects);
+        let end_cost = start_cost.saturating_add(tx_cost);
 
         // Allow tx if it's within budget.
-        if start_cost.saturating_add(tx_cost) <= self.max_accumulated_txn_cost_per_object_in_commit
-        {
+        if end_cost <= self.max_accumulated_txn_cost_per_object_in_commit {
             return None;
         }
 
-        // Allow over-budget tx if it's not above the overage limit.
-        if start_cost <= self.max_accumulated_txn_cost_per_object_in_commit
-            && start_cost.saturating_add(tx_cost)
-                <= self
-                    .max_accumulated_txn_cost_per_object_in_commit
-                    .saturating_add(self.max_txn_cost_overage_per_object_in_commit)
-        {
+        // Allow over-budget tx if it's not above the overage limits.
+        let burst_limit = self
+            .max_accumulated_txn_cost_per_object_in_commit
+            .saturating_add(self.allowed_txn_cost_overage_burst_per_object_in_commit);
+        let absolute_limit = self
+            .max_accumulated_txn_cost_per_object_in_commit
+            .saturating_add(self.max_txn_cost_overage_per_object_in_commit);
+        if start_cost <= burst_limit && end_cost <= absolute_limit {
```

Comment:
is the check on line 166 superfluous now? it seems like the only time the end cost is relevant is when checking the absolute limit.

This also fits with what we discussed, where we split the constants into the debt ceiling and the payback rate. The per commit constant doesn't seem to be needed here

Reply:
you're right, removed the superfluous check